File: mobile/app/src/main/java/com/praeter/ui/signup/license/LicenseAgreementActivity.java
package com.praeter.ui.signup.license;

import android.os.Build;
import android.os.Bundle;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.praeter.R;
import com.praeter.ui.base.BaseActivity;

import static android.view.WindowManager.LayoutParams.FLAG_FULLSCREEN;

public class LicenseAgreementActivity extends BaseActivity<LicenseAgreementView> {


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        setContentView(R.layout.activity_license_agreement);
        super.onCreate(savedInstanceState);
    }


    @Override
    protected void onDestroy() {
        view.onDestroy();
        super.onDestroy();
    }
}

File: mobile/app/src/main/java/com/praeter/ui/signup/license/LicenseAgreementActivity.kt
package com.praeter.ui.signup.license

import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.praeter.R
import com.praeter.databinding.ActivityLicenseAgreementBinding
import com.praeter.navigator.Navigator
import com.praeter.ui.base.BaseActivity
import dagger.hilt.android.AndroidEntryPoint
import timber.log.Timber

@AndroidEntryPoint
class LicenseAgreementActivity : BaseActivity() {
    private var _viewBinding: ActivityLicenseAgreementBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _viewBinding!!

    val navigator: Navigator = Navigator(this@LicenseAgreementActivity)


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        _viewBinding = ActivityLicenseAgreementBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    fun continueClick(view: View) {
        Timber.d("onContinueButtonClicked()")
        if (!binding.cbAcceptLicense.isChecked) {
            Timber.d("checkbox not checked display toast message")
            Toast.makeText(
                this,
                getString(R.string.err_msg_license_agreement_approval_mandatory),
                Toast.LENGTH_LONG
            )
                .show()
            return
        }
        Timber.d("user agreed go to the next activity")
        goToUserFormActivity()
    }

    override fun onDestroy() {
        super.onDestroy()
        Timber.e("onDestroy()")
        _viewBinding = null
    }

    private fun goToUserFormActivity() {
        navigator.callUserFormActivity()
        finish()
    }
}